### PR TITLE
NEW: align-to-tree-mafft-iqtree

### PR DESCRIPTION
**Pull request by mikerobeson**
This pull request is simply an `iqtree` analog of the `align-to-tree-mafft-fasttree` pipeline as outlined [here](https://docs.qiime2.org/2019.7/plugins/available/phylogeny/align-to-tree-mafft-fasttree/). With the exception that the phylogeny is created via [iqtree](https://docs.qiime2.org/2019.7/plugins/available/phylogeny/iqtree/) and associated parameters instead of fasttree.

I've also edited the help text associated with the `alrt` parameter within `iqtree` and `iqtree-ultrafast-bootstrap`. That is, we do not actually allow a user to supply a `0` to the `alrt` flag for the parametric method, despite what the text says. Though it'd be nice to allow the user to supply `0` or anything above `1000`.

### CODE REVIEW

**Comment by mikerobeson** [comment]
I keep getting this error message:

> Preparing transaction: ...working... done
Verifying transaction: ...working... failed
RemoveError: 'setuptools' is a dependency of conda and cannot be removed from
conda's operating environment.
The command "conda update -q conda" failed and exited with 1 during .
Your build has been stopped.

Is this something I can fix on my end, or is this a TravisCI (or other) issue?

**Comment by thermokarst** [comment]
@mikerobeson, its not you, its us (or, maybe its travis' fault?). Looking into this now...

**Comment by mikerobeson** [comment]
Sorry, I must have fat-fingered the `close pull request` button.

**Comment by thermokarst** [comment]
Looks like travis is back in business now, too. :sweat_smile:

**Comment by mikerobeson** [comment]
Since this has been slated for the 2020.1 review, and I am planning to work up a similar pipeline for raxml as well, should I include the raxml pipeline as part of this pull request (since it is simple enough)? Keep separate)? Probably makes sense to keep separate, but I thought I should ask.  

Should we revisit how we want to handle these pipelines as per our prior discussions [here](https://github.com/qiime2/q2-phylogeny/pull/49) and [here](https://forum.qiime2.org/t/q2-phylogeny-refactor/5019)? It seems that we are good with the `align-to-tree-*` labelling.

**Comment by thermokarst** [comment]
> should I include the raxml pipeline as part of this pull request (since it is simple enough)? Keep separate)?

Let's do this separately, for now. I think one thing @ebolyen was considering was making a pipeline with branching abilities, so, one pipeline where parameters would control which tree-building algorithm is used. We are missing a bit of machinery to make that comfortable (its possible right now, but I suspect it wouldn't scale well with the myriad of parameters for raxml and iqtree).

@ebolyen, want to weigh in?

> It seems that we are good with the `align-to-tree-*` labelling.

Yeah, I think this is working well enough, for now...

**Comment by ebolyen** [comment]
Yeah, I think we are missing a None type in TypeMap, so that we could say, when alignment method is "X" or phylogeny estimator is "Y" all of these other parameters must be None. This would also give us optional outputs, which would be super handy.

No estimate on when that will happen, but I think this is the second use case in the last few weeks for it.

I could see a generic `align-to-tree` that subsumes all of these pipelines in the future.

File: q2_phylogeny/plugin_setup.py [hunk 1]
```diff
@@ -476,3 +474,94 @@
...
                  )
 )
+
+plugin.pipelines.register_function(
+    function=q2_phylogeny.align_to_tree_mafft_iqtree,
+    inputs={
+        'sequences': FeatureData[Sequence],
+    },
+    parameters={
+        'n_threads': Int % Range(0, None),
```
Comment (thermokarst) [inline q2_phylogeny/plugin_setup.py]: We now have a slightly cleaner mechanism for doing the "auto" business:

```suggestion
        'n_threads': Int % Range(1, None) | Str % Choices(['auto']),
```

This change relies on https://github.com/qiime2/q2-alignment/pull/65 & https://github.com/qiime2/q2-phylogeny/pull/64

File: q2_phylogeny/tests/test_align_to_tree_mafft_iqtree.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        self.align_to_tree_mafft_iqtree = self.plugin.pipelines[
+                                    'align_to_tree_mafft_iqtree']
+
+        input_sequences_fp = self.get_data_path('dna-sequences-1.fasta')
+        self.input_sequences = Artifact.import_data('FeatureData[Sequence]',
+                                                    input_sequences_fp)
+
+    def test_execution(self):
+        # Does it run?
+        self.align_to_tree_mafft_iqtree(self.input_sequences)
```
Comment (thermokarst) [inline q2_phylogeny/tests/test_align_to_tree_mafft_iqtree.py]: Can you add in a `self.assertTrue(True)` right after this? Just as a simply "checkpoint" to ensure we are actually testing a thing.

File: q2_phylogeny/plugin_setup.py (hunk 1 above)
Comment (mikerobeson) [reply q2_phylogeny/plugin_setup.py]: Okay great! Done.

File: q2_phylogeny/tests/test_align_to_tree_mafft_iqtree.py (hunk 2 above)
Comment (mikerobeson) [reply q2_phylogeny/tests/test_align_to_tree_mafft_iqtree.py]: Done.

**Comment by thermokarst** [comment]
Hey @ebolyen, can you take a quick pass through this to make sure I didn't miss anything? Thanks!

**Pull request merged**